### PR TITLE
[UX] Update Russian keyboard layout

### LINKED ISSUE #5953: Contribution: Renewed Russian-Kazakh Keyboard
* KOReader version: tested on 2020.03.01
* Device: Kindle Paperwhite 5.9.7

Dear KOReader Developers and Community,
First of all, it was such a touching moment to realize that current KOReader build for Kindle uses the Russian-Kazakh keyboard variant I [suggested](https://github.com/koreader/koreader/issues/2014) for Kindle Touch about 4 years ago.

I am happy to see so much improvement in keyboard's frontend like popups, selecting input languages and resizing the keys, that it motivated me to edit this keyboard again :D
Here is the archive with layout and its popup file: 
[koreader_new_RU_keyboard.zip](https://github.com/koreader/koreader/files/4332318/koreader_new_RU_keyboard.zip)

I will be happy if you find this keyboard useful enough to include it in the April release.
Please let me know if something should be changed.

1) Russian layout looks more natural now, with only popup keys for ё and ъ:
Keyboard1

2) Kazakh letters moved from umlaut-layout to popups of the main layout.
Keyboard2_Kazakh

3) Umlaut section is now a full-blown Greek alphabet + math symbols (overall I feel that 8 layers of keyboard is a little to much for a keyboard, but I decided to replace the keyboard completely now that I started).
Keyboard4_Greek

4) Arrow keys in Sym section are now aligned as on a PC keyboard.
Keyboard3_Sym

5) The keyboard files are edited in the most simple and clear way without any workarounds as I've seen in some other keyboard adaptations. I hope this makes it better to read and modify if needed.

Thank you! I really appreciate everything you are doing with KOReader and would like to contribute even more!

### PR DESCRIPTION
**Pull request by mergen3107**
Closes #5953

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5957)

### CODE REVIEW

File: frontend/ui/data/keyboardlayouts/ru_keyboard.lua [hunk 1]
```diff
@@ -9,70 +9,102 @@ local _je_ = ru_popup._je_
 local _Ye_ = ru_popup._Ye_
 local _ye_ = ru_popup._ye_
 
+-- the Russian soft/hard sign
```
Comment (Frenzie) [inline frontend/ui/data/keyboardlayouts/ru_keyboard.lua]:
```suggestion
-- the Russian soft/hard sign
```

File: frontend/ui/data/keyboardlayouts/ru_keyboard.lua [hunk 2]
```diff
@@ -9,70 +9,102 @@ local _je_ = ru_popup._je_
 local _Ye_ = ru_popup._Ye_
 local _ye_ = ru_popup._ye_
 
+-- the Russian soft/hard sign 
+local _SH_ = ru_popup._SH_
```
Comment (Frenzie) [inline frontend/ui/data/keyboardlayouts/ru_keyboard.lua]:
```suggestion
local _SH_ = ru_popup._SH_
```

File: frontend/ui/data/keyboardlayouts/ru_keyboard.lua [hunk 3]
```diff
@@ -9,70 +9,102 @@ local _je_ = ru_popup._je_
 local _Ye_ = ru_popup._Ye_
 local _ye_ = ru_popup._ye_
 
+-- the Russian soft/hard sign 
+local _SH_ = ru_popup._SH_ 
+local _sh_ = ru_popup._sh_
+
+-- Kazakh Cyrillic letters: ә і ң ғ ү ұ қ ө һ
+local _KA_ = ru_popup._KA_
```
Comment (Frenzie) [inline frontend/ui/data/keyboardlayouts/ru_keyboard.lua]:
```suggestion
local _KA_ = ru_popup._KA_
```

File: frontend/ui/data/keyboardlayouts/ru_keyboard.lua [hunk 4]
```diff
@@ -9,70 +9,102 @@ local _je_ = ru_popup._je_
...
+            { _KU_,   _ku_,   "]",    "2",    "Ε",    "ε",    "²",    "2", },
+            { _KK_,   _kk_,   "{",    "3",    "Ρ",    "ρ",    "³",    "3", },
+            { _Ye_,   _ye_,   "}",    "4",    "Τ",    "τ",    "⁴",    "4", }, --
+            { _KN_,   _kn_,   "#",    "5",    "Υ",    "υ",    "⁵",    "5", }, --
+            { _KG_,   _kg_,   "%",    "6",    "Θ",    "θ",    "⁶",    "6", }, --
+            { "Ш",    "ш",    "^",    "7",    "Ι",    "ι",    "⁷",    "7", }, --
+            { "Щ",    "щ",    "*",    "8",    "Ο",    "ο",    "⁸",    "8", }, --
+            { "З",    "з",    "+",    "9",    "Π",    "π",    "⁹",    "9", }, --
+            { _KH_,   _kh_,   _eq,    "0",    "²",    "√",    "⁰",    "0", }, --
+        },
```
Comment (Frenzie) [inline frontend/ui/data/keyboardlayouts/ru_keyboard.lua]:
```suggestion
        },
```

File: frontend/ui/data/keyboardlayouts/ru_keyboard.lua [hunk 5]
```diff
@@ -9,70 +9,102 @@ local _je_ = ru_popup._je_
...
-            { "Г",    "г",    "‰",    ")",    "Ң",    "ң",    "7",    "¾", },
-            { "Ш",    "ш",    "|",   "\\",    "Ғ",    "ғ",    "8",    "©", },
-            { "Щ",    "щ",    "?",    "/",    "Х",    "х",    "9",    "®", },
-            { "З",    "з",    "~",    "`",    "Ъ",    "ъ",    "0",    "™", },
-        },
+            { "Й",    "й",    "'",    "`",    "∫",    "∂",    "∴",    "`", },
+            { "Ц",    "ц",    "[",    "1",    "ς",    "ς",    "¹",    "1", },
+            { _KU_,   _ku_,   "]",    "2",    "Ε",    "ε",    "²",    "2", },
+            { _KK_,   _kk_,   "{",    "3",    "Ρ",    "ρ",    "³",    "3", },
+            { _Ye_,   _ye_,   "}",    "4",    "Τ",    "τ",    "⁴",    "4", }, --
```
Comment (Frenzie) [inline frontend/ui/data/keyboardlayouts/ru_keyboard.lua]: Btw, what's with all the comments at the end of the line? I'd overlooked those.

**Comment by mergen3107** [comment]
I just used these comments to keep track of replacement positions. You can remove them if you want

**Pull request merged**